_includes/social-share/socialSharing.js

```diff
@@ -4,118 +4,53 @@
  * Forces the share links to open in a new and sized
  * window.
  */
-
-function shareOnFacebook () {
-	window.open(
-		'http://www.facebook.com/sharer.php?u={{ site.url }}{{ page.url }}',
-		'shareWindow',
-		'location=1,toolbar=1,menubar=1,resizable=1,width=200,height=200'
-	)
-	return false
-}
-
-function shareOnTwitter () {
-	window.open(
-		'https://twitter.com/share?url={{ site.url }}{{ page.url }}',
-		'shareWindow',
-		'location=1,toolbar=1,menubar=1,resizable=1,width=400,height=300'
-	)
-	return false
-}
-
-function shareOnGooglePlus () {
-	window.open(
-		'https://plus.google.com/share?url={{ site.url }}{{ page.url }}',
-		'shareWindow',
-		'location=1,toolbar=1,menubar=1,resizable=1,width=400,height=300'
-	)
-	return false
-}
-
-function shareOnLinkedIn () {
-	window.open(
-		'http://www.linkedin.com/shareArticle?mini=true&amp;url={{ site.url }}{{ page.url }}',
-		'shareWindow',
-		'location=1,toolbar=1,menubar=1,resizable=1,width=400,height=300'
-	)
-	return false
+function openNewWindow(element, width, height) {
+  window.open(
+    element.getAttribute('href'),
+    'shareWindow',
+    'location=1,toolbar=1,menubar=1,resizable=1,width=' + (width || 400) + ',height=' + (height || 400)
+  )
 }
 
 /**
  * Handles the scroll event.
  */
-function actOnScroll () {
-	if (!isSmallScreen()) {
-		var element = document.getElementById('social-share-container')
-		if (document.body.scrollTop >= 200 || document.documentElement.scrollTop >= 200) {
-			if (!timeToHide()) {
-				element.style.display = 'block'
-			} else {
-				element.style.display = 'none'
-			}
-		} else {
-			element.style.display = 'none'
-		}
-	} else {
-		hideBottomShareButtonOnScrollDown()
-	}
-}
+function actOnScroll() {
+  var socialShareContainer = document.getElementById('social-share-container')
 
-/**
- * Hides the share buttons of a small
- * screen device.
- */
-function hideBottomShareButtonOnScrollDown () {
-	var previous = window.scrollY
-	var element = document.getElementById('bottom-social-share-container')
-	window.addEventListener('scroll', function () {
-		if (window.scrollY > previous) {
-			element.className = 'slide-down'
-			//element.style.display = 'none'
-		} else {
-			element.className = 'slide-up'
-			//element.style.display = 'block'
-		}
-		previous = window.scrollY
-	})
+  // Skip if the links are not floating
+  if (window.getComputedStyle(socialShareContainer).getPropertyValue('position') !== 'fixed') {
```

Comment:
@danielepolencic I don't like that everything is in one method! Also, I didn't use getComputedStyle() on purpose as impact performance since JS has to process this on every scroll.